app/controllers/answers_controller.rb

```diff
@@ -0,0 +1,29 @@
+class AnswersController < ApplicationController
+
+  def new
+    @answer = Answer.new
+    @answer.question_id = @question.id
+  end
+
+  def create
+    @answer = Answer.new(answer_params)
+    @answer.question_id = params[:question_id]
+    @answer.save
+    redirect_to question_path(@answer.question)
+  end
+
+  private
+
+    def answer_params
+      params.require(:answer).permit(:user_id, :question_id, :response)
+    end
+
+    def set_answer
```

Comment:
These functions aren't being used - do you want a before_action to go with them? Another way to do this is with lazy execution, create a function like this:

```
def answer
  @answer ||= begin
    answer = params[:id] ? Answer.find(params[:id]) : Answer.new
    if answer_params && answer_params.length > 0
      answer.update_attributes(answer_params)
   end
  end
end
```